scipy/stats/_continuous_distns.py

```diff
@@ -4220,10 +4220,6 @@ def fit(self, data, *args, **kwds):
 
 
 def _check_fit_input_parameters(data, args, kwds, fixed_param):
-    if len(args) > 0:
-        raise TypeError("Too many arguments.")
-
-    _remove_optimizer_parameters(kwds)
```

Comment:
This was added in the addtion of the Laplace MLE fit method, but I think these can be removed. 

-  `if len(args) > 0: .. ` is not backwards compatable
-  `_remove_optimizer_parameters(kwds)`: We may want to keep the optimizer parameters as some branches of our overriden fit methods need to call the super method.

Reply:
Instead, let's just perform the check we actually want to perform here. I think all this `pop`ing of `kwds` is really confusing.  All of this can be replaced with one line: 
```
{'loc', 'scale', 'optimizer'}.issuperset(kwds)
```
But is that actually what we want to do? What if the user passes in `b` as a guess?